src/main/java/org/opensearch/plugin/insights/core/exporter/LocalIndexExporter.java

```diff
@@ -113,4 +119,47 @@ public void close() {
     private String getDateTimeFromFormat() {
         return indexPattern.print(DateTime.now(DateTimeZone.UTC));
     }
+
+    /**
+     * Set local index exporter data retention period
+     *
+     * @param deleteAfter the number of days after which Top N local indices should be deleted
+     */
+    public void setDeleteAfter(final int deleteAfter) {
+        this.deleteAfter = deleteAfter;
+    }
+
+    /**
+     * Delete Top N local indices older than the configured data retention period
+     */
+    public void deleteExpiredIndices(final Map<String, IndexMetadata> indexMetadataMap) {
+        long expirationMillisLong = System.currentTimeMillis() - TimeUnit.DAYS.toMillis(deleteAfter);
```

Comment:
One thought for testing: we can abstract the "finding expired indices" logic from deleteExpiredIndices into a seperate function like
```java
public List<String> findExpiredIndices(final Map<String, IndexMetadata> indexMetadataMap) {
    List<String> expiredIndices = new ArrayList<>();
    long expirationMillisLong = System.currentTimeMillis() - TimeUnit.DAYS.toMillis(deleteAfter);
    
    for (IndexMetadata indexMetadata : indexMetadataMap.values()) {
        String indexName = indexMetadata.getIndex().getName();
        if (matchesPattern(indexName, indexPattern) && 
            indexMetadata.getCreationDate() <= expirationMillisLong) {
            expiredIndices.add(indexName);
        }
    }
    return expiredIndices;
}
```
`deleteExpiredIndices` can then call this function to get the expired indices, we can also add more unit tests on this function by mocking some expired indexMetadata

Reply:
I have added a UT for `deleteExpiredIndices` with mock index metadata. I will push it once I finish the remaining comments.